README.md

```diff
@@ -166,56 +168,57 @@ We provide a high performance [packet generator](simulator) to emulate the RRU.
+Agora supports a 64-antenna Faros base station as RRU and Iris UE devices. Both are commercially available from [Skylark Wireless](https://skylarkwireless.com) and are used in the [POWER-RENEW PAWR testbed](https://powderwireless.net/).\
+Both Faros and Iris have their roots in the [Argos massive MIMO base station](https://www.yecl.org/argos/), especially [ArgosV3](https://www.yecl.org/argos/pubs/Shepard-MobiCom17-Demo.pdf). 
+Agora also supports USRP-based RRU and UEs. We use command line variables of `cmake`\
+to switch between emulated RRU and real RRU. We use `-DUSE_AGROS` for Faros RRU and Iris UEs, and `-DUSE_UHD` for USRP-based RRU and UEs. 
 
-We use command line variables of cmake to switch between emulated RRU and real RRU. 
-We use `-DUSE_AGROS` for Faros RRU and Iris UEs, and `-DUSE_UHD` for USRP-based RRU and UEs. 
-
-Agora supports both uplink and downlink with real RRU and UEs. 
-For downlink, a reference node outside the array (and synchronized) is required for reciprocity calibration.
 We recommend using one server for controlling the RRU and running Agora, 
 and another server for controlling the UEs and running the UE code.
-Below we describe how to get the uplink and downlink demos working.
+
+The hardware connected to the server can be discovered using the [pyfaros](https://github.com/skylarkwireless/pyfaros) tool.
+
+The command `python3 -m pyfaros.discover --json-out` outputs a JSON file used as a configuration input for Agora as explained below.
+
+Agora supports both uplink and downlink with real RRU and UEs.
+
+For downlink, a reference node outside the array (and synchronized) is required for reciprocity calibration.
+It should also be indicated in the hardware JSON file.
+
+We describe how to get the uplink and downlink demos working. Below XX can be replaced with either `ul` and `dl`.
  * Rebuild the code on both servers for RRU side the UE side.
     * For Faros RRU and Iris UEs, pass `-DUSE_ARGOS=on -DUSE_UHD=off` to cmake
     * For USRP-based RRU and UEs, pass `-DUSE_ARGOS=off -DUSE_UHD=on` to cmake 
     * Run `make -j` to recompile the code.
  * Run the UE code on the server connected to the Iris UEs
-   * Modify `data/user-iris-serials.txt` by adding serials of two client Irises
-     from your setup.
-   * Run `./build/data_generator --conf_file data/ue-ul-hw.json` to generate required data files. 
-   * Run `./build/user --conf_file data/ue-ul-hw.json`.
-   * For downlink, use `data/ue-dl-hw.json` in the last two steps.
+   * Use pyfaros tool to generate a 
+   * Modify `data/topology.json` by adding/removing serials of client Irises you'd like to include
+     from your setup. This file can be generated using the `pyfaros` tool as described above.
+   * Run `./build/data_generator --conf_file data/XX-hw.json` to generate required data files. 
```

Comment:
Maybe we should mention what XX represents here?

Reply:
I explain XX above it.